targets/targets.json

```diff
@@ -8778,6 +8777,65 @@
         "mbed_ram_start"        : "0x20000000",
         "mbed_ram_size"         : "0x8000"
     },
+    "NU_PFM_M2351_NS": {
+        "inherits": ["NSPE_Target", "NU_PFM_M2351_CM"],
+        "core": "Cortex-M23-NS",
+        "tfm.level": 1,
+        "extra_labels_add": [
+            "M23_NS",
+            "PSA",
+            "TFM",
+            "NU_PREBUILD_SECURE"
+        ],
+        "macros_add": [
+            "MBED_TZ_DEFAULT_ACCESS=1",
+            "CMSIS_NVIC_VIRTUAL",
+            "MBEDTLS_PSA_CRYPTO_C"
+        ],
+        "components_add": ["FLASHIAP"],
+        "post_binary_hook": {"function": "M2351Code.merge_secure"},
+        "secure_image_filename": "tfm.hex",
+        "overrides": {
+            "secure-rom-start"      : "0x0",
```

Comment:
Why the non-secure side needs to know about the fact from where does the secure side start?

Reply:
Yes, non-secure side needn't. However, in rare case like `mbed_hal-spm` test, secure memory is intentionally accessed to trigger [security fault](https://github.com/ARMmbed/mbed-os/blob/eeb033f8148a87659f3c084ec0b69e59ba9ac78e/TESTS/mbed_hal/spm/main.cpp#L102), though this test is filtered out because SecureFault cannot remap.